### ISSUE TITLE
'Sort by Progress' broken for Qt

### ISSUE THREAD

**Issue by OscarCunningham** [transmission/transmission#3796]
I'm running transmission-qt version 3.00 (from Debian testing) and seeing that my seeding torrents are in the wrong order when sorting by progress. My torrents which haven't finished downloading yet are in the right order, but those which are just seeding are not sorted by ratio (nor by total bytes seeded).

I believe the problem is in `compareSeedRatio` on lines 59 & 60 of qt/Torrent.cc.
```
    auto const a = getSeedRatioLimit();
    auto const b = that.getSeedRatioLimit();
```
This looks like it gets the ratio limit, rather than the ratio itself. Since I use a global ratio limit of 2.0, this will be the same for all torrents, so they end up defaulting to sorting by queue order. I confirmed this by manually setting the limit for a torrent to 1.99 and 2.01, and it did indeed go to the top and bottom of the list.

**Comment by ckerr** [comment]
Yep, looks like `compareSeedRatio()` should use a combo of `Torrent::getSeedRatioLimit()` + `Torrent::ratio()`

**Comment by OscarCunningham** [comment]
Fixed by my PR #3814.
